### DISCUSSION TITLE
@babel/types How do I create a member access like this: foo["bar"]

### DISCUSSION THREAD

**Discussion by bali182** [babel/babel#13469] [Q&A] [answered]
I'd like to create an ast for an expression like this:

```
foo["bar"]
```

Based on the parsed data here: https://astexplorer.net/#/gist/ce4614cfa118a80267ae7e931a96093a/7caa17af416d9d41fa70828ac128cf54b24b29cb

I would assume I can do this (typescript typings also allow this):

```ts
import { identifier, memberExpression, stringLiteral } from '@babel/types'

describe('test foo["bar"] ast creation', () => {
  it('should generate memberExpression', () => {
    expect(() => memberExpression(identifier('foo'), stringLiteral('bar'))).not.toThrowError()
  })
})
```

But I get an error saying: 

> TypeError: Property property of MemberExpression expected node to be of a type ["Identifier","PrivateName"] but instead got "StringLiteral"

So it looks like `memberExpression` only takes those 2 arg types as second parameters. 

Whats the proper way of doing this?

**Comment by JLHwung** [comment, marked as answer]
You need to pass the third argument to instruct babel it is a _computed_ member expression:

```js
import { identifier, memberExpression, stringLiteral } from '@babel/types'

memberExpression(identifier('foo'), stringLiteral('bar'), true)
```

See docs here: https://babeljs.io/docs/en/babel-types#memberexpression

**Reply by bali182** [reply]
Thanks very much, that did the trick!
